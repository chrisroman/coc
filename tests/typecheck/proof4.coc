let idT = [a:*][x:a]a in
Theorem id = idT with
Proof (\a:*)(\x:a)x;
id
