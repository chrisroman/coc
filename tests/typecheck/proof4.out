let idT = [a: *] [x: a] a in 
Theorem id = idT with 
	Proof (λa: *) (λx: a) x;
id
{idT} typecheck_single_term [a: *] [x: a] a
Type: *
typecheck_single_term [a: *] [x: a] a
Type: *
typecheck_single_term (λa: *) (λx: a) x
Type: [a: *] [x: a] a
typecheck_single_term (λa: *) (λx: a) x
Type: [a: *] [x: a] a
Value: (λa: *) (λx: a) x
